### PR TITLE
solana fee estimation fixes

### PR DESCRIPTION
**Pull request by detherminal**
:)

### CODE REVIEW

File: lib/wallets/wallet/impl/solana_wallet.dart [hunk 1]
```diff
@@ -66,10 +65,8 @@ class SolanaWallet extends Bip39Wallet<Solana> {
         recipientAccount: pubKey,
         lamports: transferAmount.raw.toInt(),
       ),
-    ]).compile(
-      recentBlockhash: latestBlockhash!.value.blockhash,
-      feePayer: pubKey,
-    );
+      ComputeBudgetInstruction.setComputeUnitPrice(microLamports: 6000),
```
Comment (sneurlax) [inline lib/wallets/wallet/impl/solana_wallet.dart:68]: Why 6000 here?

Comment (detherminal) [reply lib/wallets/wallet/impl/solana_wallet.dart:68]: transfer instruction is 5000 lamports and compute unit budget instruction is 1000 lamports, total of 6k lamports

Comment (julian-CStack) [reply lib/wallets/wallet/impl/solana_wallet.dart:68]: > transfer instruction is 5000 lamports and compute unit budget instruction is 1000 lamports, estimated total of 6k lamports

Include this explanation in a comment please so the seemingly random 6000 is documented in code

**Comment by detherminal** [comment]
Current approach of fee:
- Get only necessary transfer fee
- Add fee on top of base if it is higher than it
- Limit the tx to not use more than the specified amount

Also fixed the sent to self bug, it is working correctly now

**Review by sneurlax — APPROVED** [review]
Looks good to me!

**Pull request merged**